### PR TITLE
Added command line options to set the host and external_host.

### PR DESCRIPTION
**Pull request by asajeffrey**
In the Servo CI, it would be nice to run the wpt tests with `external_host` set to `localhost`, since that exposes some corner cases (in particular, some cases where fetch is not throwing security errors).

At the moment, setting `host` and `external_host` requires editing `wptrunner/browsers/servo.py`. This PR allows them to be set from the command line.

The Servo issue is https://github.com/servo/servo/issues/15232.

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/wptrunner/231)

### CODE REVIEW

**Comment by asajeffrey** [comment]
cc @jgraham

File: wptrunner/wptrunner.py [hunk 1]
```diff
@@ -148,6 +148,9 @@ def run_tests(config, test_paths, product, **kwargs):
 
         kwargs["pause_after_test"] = get_pause_after_test(test_loader, **kwargs)
 
+        env_options["host"] = kwargs.get("host") or env_options["host"]
+        env_options["external_host"] = kwargs.get("external_host") or env_options["external_host"]
```
Comment (andreastt) [inline wptrunner/wptrunner.py]: This could be kwargs.get("host", env_options["host"]). If kwargs.get("host") returns a falsy value (false, 0, "", &c.) that will select env_options["host"], but using the dict.get(key, fallback) method will perform a proper None check.

Comment (asajeffrey) [inline wptrunner/wptrunner.py]: Done.

Comment (asajeffrey) [inline wptrunner/wptrunner.py]: Actually, this doesn't work, because when the options are parsed, `kwargs["host"]` is always set, it is just set to `None` if the `--host` is not present. So I think the original code was doing the right thing.

**Comment by jgraham** [comment]
So, a few things:

 * Running tests on localhost is a bad idea because it's a special domain from the point of view of various security features (e.g. https only things often run from http://localhost).
* Servo really should make the process separation consider scheme and port. Otherwise the security benefits of process separation seem to be compromised.
* If that isn't possible for some reason, we should add a new top level domain to web-platform-tests rather than having servo hack around it.
* In general the configuration of wptrunner is a mess and this doesn't make it better ;)

So I think I have an order of preferences: servo to change, wpt to change, wptrunner to change. I'll reconsider this if the other options are viable for some reason.

**Comment by asajeffrey** [comment]
@jgraham agreed, Servo should base the assignment to processes on eTLD+1 origin, not just host. If we did this, the problem is that tests are using `HTTP_REMOTE_ORIGIN`, (set in `common/get-host-info.sub,js`) which is similar-origin, rather than `HTTP_REMOTE_ORIGIN_WITH_DIFFERENT_PORT`.

**Comment by asajeffrey** [comment]
On thinking about it, the simplest change would be in `get-host-info.sub.js` to set:
```js
  var REMOTE_HOST = (ORIGINAL_HOST === '127.0.0.1') ? '127.0.0.2' : '127.0.0.1';
```
Does this seem sensible? If not, we probably want to add a new TLD to wpt.

**Comment by asajeffrey** [comment]
Discussed this with @jgraham a while back, and decided that the long-term fix is for wpt to support more than one domain, and for the servo-specific dissimilar-origin tests to sit in `/mozilla`. So this patch isn't needed.
